src/main/java/com/gmail/val59000mc/listeners/TeleportListener.java

```diff
@@ -63,6 +63,28 @@ public void onPlayerPortalEvent (PlayerPortalEvent event){
 			}
 		}
 	}
+	
+	@EventHandler
+	public void overworldTeleport(PlayerChangedWorldEvent event) {
+		
+		GameManager gm = GameManager.getGameManager();
+		Player player = event.getPlayer();
+		
+		if(gm.getConfiguration().getEnableTheEnd() && event.getFrom().getName().equals(gm.getConfiguration().getTheEndUuid())) {
+			
+			UhcWorldBorder borderbounds = new UhcWorldBorder();
+			World world = Bukkit.getServer().getWorld(gm.getConfiguration().getOverworldUuid());
+			int x = RandomUtils.randomInteger((int) Math.round(-borderbounds.getCurrentSize()), (int) Math.round(borderbounds.getCurrentSize()));
+			int z = RandomUtils.randomInteger((int) Math.round(-borderbounds.getCurrentSize()), (int) Math.round(borderbounds.getCurrentSize()));
```

Comment:
Can't x and z just be 0?

Reply:
I was thinking about if they had dragon rush enabled then they would be teleported back into the end but then i realized that if they were returning to the overworld with dragon rush enabled the game is over :P

Reply:
but i'm gonna keep it random just so people don't get lava-trapped or anything i guess, your preference

Reply:
Didn't think of the trapping that could happen, so I agree a random location would be better. There is however already a method to generate a random location that is also safe (No oceans / lava lakes).

`double maxDistance = 0.9 *  gm.getWorldBorder().getCurrentSize();
Location loc = findRandomSafeLocation(world, maxDistance);`

If you could change that it's all good! :)